ITACrashCourse/flask_project
# Feature/project containerization

This pull request creates containers for the flask_project. The containers are build with using a Docker.

# Changes
- Created Dockerfile for flask application.
- Created docker-compose.yml file for our flask_app and postgres instance.
- Configured project, install dependencies.
- Delete old postgres docker.

# How to test
1. Clone repo (or git pull)
2. Switch to feature/project-containerization branch
3. Create .env file like this in main directory.

`IPQS_SECRET_KEY='your_secret_key'
DB_CONTAINER_NAME='xxxxxxxxxx'
POSTGRES_USER='xxxxxxxxxxxxx'
POSTGRES_PASSWORD='xxxxxxxxx'
POSTGRES_DB='xxxxxxxxxxxxxxxx'
POSTGRES_PORT='xxxxxxxxxx'
DATABASE_URL="postgresql://${POSTGRES_USER}:${POSTGRES_PASSWORD}@postgresql_container:5432/${POSTGRES_DB}"`

5. Run containers via docker-compose up -d --build command.
6. Verify that the project is running correctly when two containers is running.
![image](https://user-images.githubusercontent.com/87909623/217365987-38903e94-f699-4452-a15f-d0c352c7a0be.png)

5.1. Visit 127.0.0.1:5000 in your browser, you should see Hello paragraph.
![image](https://user-images.githubusercontent.com/87909623/217365937-5a5ad31d-d304-4931-b940-8103a42c27e1.png)

5.2. Open flask_app container shell and type:

- flask --app flask_project shell
- db.create_all()
![image](https://user-images.githubusercontent.com/87909623/217365869-b1a15950-21bd-4ac9-972f-3432ac87abff.png)

5.2. Check second container to check if it is a postgres instance, then open shell and type:
- psql -U <your_username> <postgres_db>
- \dt;
You should see something like:
![image](https://user-images.githubusercontent.com/87909623/217365819-b53344cf-d37a-444d-9cba-14e13a092b55.png)
